Build failure: evolution-ews

### Nixpkgs version

- Unstable (25.11)

### Steps to reproduce

`nix build 'nixpkgs#evolution-ews'`

### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

https://hydra.nixos.org/build/313146151

### Relevant log output

```console
[ 45%] Building C object src/EWS/common/CMakeFiles/evolution-ews.dir/e-ews-query-to-restriction.c.o
/build/evolution-ews-3.56.2/src/EWS/common/e-ews-query-to-restriction.c: In function 'message_func_relative_months':
/build/evolution-ews-3.56.2/src/EWS/common/e-ews-query-to-restriction.c:856:35: error: implicit declaration of function 'camel_folder_search_util_add_months'; did you mean 'camel_search_util_add_months'? [-Wimplicit-function-declaration]
  856 |                 r->value.number = camel_folder_search_util_add_months (time (NULL), argv[0]->value.number);
      |                                   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |                                   camel_search_util_add_months
/build/evolution-ews-3.56.2/src/EWS/common/e-ews-query-to-restriction.c:856:35: warning: nested extern declaration of 'camel_folder_search_util_add_months' [-Wnested-externs]
make[2]: *** [src/EWS/common/CMakeFiles/evolution-ews.dir/build.make:255: src/EWS/common/CMakeFiles/evolution-ews.dir/e-ews-query-to-restriction.c.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:1111: src/EWS/common/CMakeFiles/evolution-ews.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[ 46%] Building C object src/Microsoft365/calendar/CMakeFiles/ecalbackendmicrosoft365.dir/e-cal-backend-m365-factory.c.o
[ 46%] Building C object src/Microsoft365/calendar/CMakeFiles/ecalbackendmicrosoft365.dir/e-cal-backend-m365-recur-blob.c.o
[ 47%] Building C object src/Microsoft365/calendar/CMakeFiles/ecalbackendmicrosoft365.dir/e-cal-backend-m365-utils.c.o
[ 47%] Linking C shared module libecalbackendmicrosoft365.so
[ 47%] Built target ecalbackendmicrosoft365
make: *** [Makefile:136: all] Error 2
```

### Additional context

https://github.com/NixOS/nixpkgs/pull/458325

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.17.7, NixOS, 25.11 (Xantusia), 25.11.20251112.c5ae371`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.31.2`
 - channels(root): `"nixos"`
 - nixpkgs: `/nix/store/c9asdiqw1rf5qwdjz26f57hp68ppb8fc-source`


### Notify maintainers

@dasj19 @jtojnar 
 
---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/ev/evolution-ews/hardcode-gsettings.patch
+++ b/pkgs/by-name/ev/evolution-ews/hardcode-gsettings.patch
@@ -1,8 +1,8 @@
 diff --git a/src/EWS/camel/camel-ews-utils.c b/src/EWS/camel/camel-ews-utils.c
-index 32817df..c8b3f2e 100644
+index 9dcc5b1..9588be3 100644
 --- a/src/EWS/camel/camel-ews-utils.c
 +++ b/src/EWS/camel/camel-ews-utils.c
-@@ -1550,7 +1550,7 @@ ews_utils_save_category_changes (GHashTable *old_categories, /* gchar *guid ~> C
+@@ -1562,7 +1562,7 @@ ews_utils_save_category_changes (GHashTable *old_categories, /* gchar *guid ~> C
  	gboolean changed = FALSE;
  
  	/* cannot save, when evolution is not installed */
@@ -11,7 +11,7 @@ index 32817df..c8b3f2e 100644
  		return FALSE;
  
  	if (!old_categories || !new_categories)
-@@ -1558,7 +1558,18 @@ ews_utils_save_category_changes (GHashTable *old_categories, /* gchar *guid ~> C
+@@ -1570,7 +1570,18 @@ ews_utils_save_category_changes (GHashTable *old_categories, /* gchar *guid ~> C
  
  	evo_labels = g_ptr_array_new_full (5, g_free);
  
@@ -32,10 +32,10 @@ index 32817df..c8b3f2e 100644
  
  	for (ii = 0; strv && strv[ii]; ii++) {
 diff --git a/src/Microsoft365/camel/camel-m365-store.c b/src/Microsoft365/camel/camel-m365-store.c
-index 7374c36..dbf9099 100644
+index a4997c7..94be919 100644
 --- a/src/Microsoft365/camel/camel-m365-store.c
 +++ b/src/Microsoft365/camel/camel-m365-store.c
-@@ -305,7 +305,7 @@ m365_store_save_category_changes (GHashTable *old_categories, /* gchar *id ~> Ca
+@@ -311,7 +311,7 @@ m365_store_save_category_changes (GHashTable *old_categories, /* gchar *id ~> Ca
  	gboolean changed = FALSE;
  
  	/* cannot save, when evolution is not installed */
@@ -44,7 +44,7 @@ index 7374c36..dbf9099 100644
  		return FALSE;
  
  	if (!old_categories || !new_categories)
-@@ -313,7 +313,18 @@ m365_store_save_category_changes (GHashTable *old_categories, /* gchar *id ~> Ca
+@@ -319,7 +319,18 @@ m365_store_save_category_changes (GHashTable *old_categories, /* gchar *id ~> Ca
  
  	evo_labels = g_ptr_array_new_full (5, g_free);
  

--- a/pkgs/by-name/ev/evolution-ews/package.nix
+++ b/pkgs/by-name/ev/evolution-ews/package.nix
@@ -23,11 +23,11 @@
 
 stdenv.mkDerivation rec {
   pname = "evolution-ews";
-  version = "3.56.2";
+  version = "3.58.1";
 
   src = fetchurl {
     url = "mirror://gnome/sources/${pname}/${lib.versions.majorMinor version}/${pname}-${version}.tar.xz";
-    hash = "sha256-Hrfsz5TGuGGNa0XDICNQIJ21Z8SJEIrcICM8TRn48o8=";
+    hash = "sha256-EABr14/7fWQJxdAKVYvp3BkOhZS7miuXPi7AVhMWeCA=";
   };
 
   patches = [

